### PR TITLE
snip bad authorization api dependencies

### PR DESCRIPTION
**Pull request by deads2k**
Dependencies outside of apimachinery and the upstream api objects need to be removed.  This removes several from authorization.openshift.io.

@openshift/sig-security fyi
@mfojtik

### CODE REVIEW

File: pkg/authorization/apis/authorization/helpers.go [hunk 1]
```diff
@@ -88,29 +88,61 @@ func BuildSubjects(users, groups []string, userNameValidator, groupNameValidator
...
 		subjects = append(subjects, kapi.ObjectReference{Kind: kind, Name: group})
 	}
 
 	return subjects
 }
 
-func determineUserKind(user string, userNameValidator validation.ValidateNameFunc) string {
+// duplicated from the user/validation package.  We need to avoid api dependencies on validation from our types.
+// These validators are stable and realistically can't change.
+func validateUserName(name string, _ bool) []string {
```
Comment (deads2k) [inline pkg/authorization/apis/authorization/helpers.go]: I already discussed this with @liggitt

**Comment by enj** [comment]
/lgtm

@deads2k if I understand correctly the import restrictions will prevent this from being accidentally undone?

**Comment by deads2k** [comment]
> @deads2k if I understand correctly the import restrictions will prevent this from being accidentally undone?

correct

**Pull request merged**